src/app/test_executive/common_prefix.ml

```diff
@@ -0,0 +1,60 @@
+open Core_kernel
+open Integration_test_lib
+
+module Make (Inputs : Intf.Test.Inputs_intf) = struct
+  open Inputs
+  open Engine
+
+  type network = Network.t
+
+  type node = Network.Node.t
+
+  type dsl = Dsl.t
+
+  let config =
+    let open Test_config in
+    { default with
+      block_producers=
+        [ {balance= "1000"; timing= Untimed}
+        ; {balance= "1000"; timing= Untimed}
+        ; {balance= "1000"; timing= Untimed}
+        ; {balance= "1000"; timing= Untimed}
+        ; {balance= "1000"; timing= Untimed} ] }
+
+  let take xall n =
+    match xall with
+    | [] ->
+        []
+    | x :: xs ->
+        if n = zero then [] else x :: take xs (n - 1)
+
+  let check_common_prefixes prefixes =
+    let recent_chains =
+      List.map prefixes ~f:(take prefixes 5 |> Hash_set.of_list (module String))
```

Comment:
it might be better to pass the prefix length to `check_common_prefixes`, perhaps with a default